Build failure: chatterino2

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

`nix build .#chatterino2`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/285863854

### Relevant log output

```console
> [ 20%] Building CXX object src/CMakeFiles/chatterino-lib.dir/controllers/commands/builtin/twitch/AddModerator.cpp.o
       > [ 21%] Building CXX object src/CMakeFiles/chatterino-lib.dir/controllers/commands/builtin/twitch/AddVIP.cpp.o
       > [ 21%] Building CXX object src/CMakeFiles/chatterino-lib.dir/controllers/commands/builtin/twitch/Announce.cpp.o
       > [ 21%] Building CXX object src/CMakeFiles/chatterino-lib.dir/controllers/commands/builtin/twitch/Ban.cpp.o
       > [ 21%] Building CXX object src/CMakeFiles/chatterino-lib.dir/controllers/commands/builtin/twitch/Block.cpp.o
       > In file included from /build/source/src/Application.cpp:42:
       > /build/source/src/providers/twitch/PubSubManager.hpp:8: fatal error: boost/asio/io_service.hpp: No such file or directory
       >     8 | #include <boost/asio/io_service.hpp>
       > compilation terminated.
       > make[2]: *** [src/CMakeFiles/chatterino-lib.dir/build.make:248: src/CMakeFiles/chatterino-lib.dir/Application.cpp.o] Error 1
       > make[2]: *** Waiting for unfinished jobs....
       > make[1]: *** [CMakeFiles/Makefile2:662: src/CMakeFiles/chatterino-lib.dir/all] Error 2
       > make: *** [Makefile:136: all] Error 2
       For full logs, run 'nix log /nix/store/g3yaqk9l913qdxkhrbfi8p0jhd21d6fh-chatterino2-2.5.2.drv'.
┏━ 1 Errors:
┃ error: builder for '/nix/store/g3yaqk9l913qdxkhrbfi8p0jhd21d6fh-chatterino2-2.5.2.drv' failed with exit code 2;
┃        last 25 log lines:
┃        > [ 17%] Building CXX object src/CMakeFiles/chatterino-lib.dir/common/QLogging.cpp.o
┃        > [ 18%] Building CXX object src/CMakeFiles/chatterino-lib.dir/common/network/NetworkCommon.cpp.o
┃        > [ 18%] Building CXX object src/CMakeFiles/chatterino-lib.dir/common/network/NetworkManager.cpp.o
┃        > [ 18%] Building CXX object src/CMakeFiles/chatterino-lib.dir/common/network/NetworkPrivate.cpp.o
┃        > [ 18%] Building CXX object src/CMakeFiles/chatterino-lib.dir/common/network/NetworkRequest.cpp.o
┃        > [ 19%] Building CXX object src/CMakeFiles/chatterino-lib.dir/common/network/NetworkResult.cpp.o
┃        > [ 19%] Building CXX object src/CMakeFiles/chatterino-lib.dir/common/network/NetworkTask.cpp.o
┃        > [ 19%] Building CXX object src/CMakeFiles/chatterino-lib.dir/controllers/accounts/Account.cpp.o
┃        > [ 19%] Building CXX object src/CMakeFiles/chatterino-lib.dir/controllers/accounts/AccountController.cpp.o
┃        > [ 20%] Building CXX object src/CMakeFiles/chatterino-lib.dir/controllers/accounts/AccountModel.cpp.o
┃        > [ 20%] Building CXX object src/CMakeFiles/chatterino-lib.dir/controllers/commands/builtin/chatterino/Debugging.cpp…
┃        > [ 20%] Building CXX object src/CMakeFiles/chatterino-lib.dir/controllers/commands/builtin/Misc.cpp.o
┃        > [ 20%] Building CXX object src/CMakeFiles/chatterino-lib.dir/controllers/commands/builtin/twitch/AddModerator.cpp.o
┃        > [ 21%] Building CXX object src/CMakeFiles/chatterino-lib.dir/controllers/commands/builtin/twitch/AddVIP.cpp.o
┃        > [ 21%] Building CXX object src/CMakeFiles/chatterino-lib.dir/controllers/commands/builtin/twitch/Announce.cpp.o
┃        > [ 21%] Building CXX object src/CMakeFiles/chatterino-lib.dir/controllers/commands/builtin/twitch/Ban.cpp.o
┃        > [ 21%] Building CXX object src/CMakeFiles/chatterino-lib.dir/controllers/commands/builtin/twitch/Block.cpp.o
┃        > In file included from /build/source/src/Application.cpp:42:
┃        > /build/source/src/providers/twitch/PubSubManager.hpp:8: fatal error: boost/asio/io_service.hpp: No such file or directory
┃        >     8 | #include <boost/asio/io_service.hpp>
┃        > compilation terminated.
┃        > make[2]: *** [src/CMakeFiles/chatterino-lib.dir/build.make:248: src/CMakeFiles/chatterino-lib.dir/Application.cpp.…
┃        > make[2]: *** Waiting for unfinished jobs....
┃        > make[1]: *** [CMakeFiles/Makefile2:662: src/CMakeFiles/chatterino-lib.dir/all] Error 2
┃        > make: *** [Makefile:136: all] Error 2
┃        For full logs, run 'nix log /nix/store/g3yaqk9l913qdxkhrbfi8p0jhd21d6fh-chatterino2-2.5.2.drv'.
┣━ Dependency Graph:
┃ ⚠ chatterino2-2.5.2 failed with exit code 2 after ⏱ 47s in buildPhase
┣━━━
┗━ ∑ ⚠ Exited after 1 build failures at 14:35:49 after 1m15s
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.9, NixOS, 25.05 (Warbler), 25.05.20250116.5df4362`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.25.3`
 - nixpkgs: `/run/current-system/nixpkgs`

### Notify maintainers

@rexim
@0Supa
@nycodeghg

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ch/chatterino2/package.nix
+++ b/pkgs/by-name/ch/chatterino2/package.nix
@@ -3,42 +3,48 @@
   callPackage,
   fetchFromGitHub,
   nix-update-script,
+  boost186,
 }:
 
-(callPackage ./common.nix { }).overrideAttrs (finalAttrs: _: {
-  pname = "chatterino2";
-  version = "2.5.2";
+(callPackage ./common.nix {
+  boost = boost186;
+}).overrideAttrs
+  (
+    finalAttrs: _: {
+      pname = "chatterino2";
+      version = "2.5.2";
 
-  src = fetchFromGitHub {
-    owner = "Chatterino";
-    repo = "chatterino2";
-    tag = "v${finalAttrs.version}";
-    hash = "sha256-nrw4dQ7QjPPMbZXMC+p3VgUQKwc1ih6qS13D9+9oNuw=";
-    fetchSubmodules = true;
-  };
+      src = fetchFromGitHub {
+        owner = "Chatterino";
+        repo = "chatterino2";
+        tag = "v${finalAttrs.version}";
+        hash = "sha256-nrw4dQ7QjPPMbZXMC+p3VgUQKwc1ih6qS13D9+9oNuw=";
+        fetchSubmodules = true;
+      };
 
-  passthru = {
-    buildChatterino = args: callPackage ./common.nix args;
-    updateScript = nix-update-script { };
-  };
+      passthru = {
+        buildChatterino = args: callPackage ./common.nix args;
+        updateScript = nix-update-script { };
+      };
 
-  meta = {
-    description = "Chat client for Twitch chat";
-    mainProgram = "chatterino";
-    longDescription = ''
-      Chatterino is a chat client for Twitch chat. It aims to be an
-      improved/extended version of the Twitch web chat. Chatterino 2 is
-      the second installment of the Twitch chat client series
-      "Chatterino".
-    '';
-    homepage = "https://github.com/Chatterino/chatterino2";
-    changelog = "https://github.com/Chatterino/chatterino2/blob/${finalAttrs.src.rev}/CHANGELOG.md";
-    license = lib.licenses.mit;
-    platforms = lib.platforms.unix;
-    maintainers = with lib.maintainers; [
-      rexim
-      supa
-      marie
-    ];
-  };
-})
+      meta = {
+        description = "Chat client for Twitch chat";
+        mainProgram = "chatterino";
+        longDescription = ''
+          Chatterino is a chat client for Twitch chat. It aims to be an
+          improved/extended version of the Twitch web chat. Chatterino 2 is
+          the second installment of the Twitch chat client series
+          "Chatterino".
+        '';
+        homepage = "https://github.com/Chatterino/chatterino2";
+        changelog = "https://github.com/Chatterino/chatterino2/blob/${finalAttrs.src.rev}/CHANGELOG.md";
+        license = lib.licenses.mit;
+        platforms = lib.platforms.unix;
+        maintainers = with lib.maintainers; [
+          rexim
+          supa
+          marie
+        ];
+      };
+    }
+  )

--- a/pkgs/by-name/ch/chatterino7/package.nix
+++ b/pkgs/by-name/ch/chatterino7/package.nix
@@ -3,10 +3,12 @@
   chatterino2,
   fetchFromGitHub,
   nix-update-script,
+  boost186,
 }:
 
 (chatterino2.buildChatterino {
   enableAvifSupport = true;
+  boost = boost186;
 }).overrideAttrs
   (
     finalAttrs: _: {

